### PR TITLE
Add algorithm asymmetric key

### PR DESCRIPTION
**Pull request by aljones15**
Adds a new property to `AsymetricKey` `algorithm` which is determined by matching the `publicKeyMultibase`  prefix to the JOSE algorithm name.

### CODE REVIEW

File: lib/AsymmetricKey.js [hunk 1]
```diff
@@ -105,6 +106,35 @@ export class AsymmetricKey {
...
+    // maps the bs58 prefix to the JOSE algorithm name
+    const algorithms = new Map([
+      ['6Mk', 'Ed25519'],
+      ['Dna', 'P-256'],
+      ['82L', 'P-384'],
+      ['2J9', 'P-521'],
+      // FIXME this prefix was not generated by an ed448 did key implementation
+      ['6GP', 'Ed448'],
+      ['Q3s', 'secp256k1']
+    ]);
```
Comment (dlongley) [inline lib/AsymmetricKey.js:130]: This should be a constant external to this function (there's no need to recreate it on every function call).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # webkms-client ChangeLog
 
+## 12.1.0 -
+
+### Added
+- AsymmetricKey now sets algorithm using the prefix of the `publicKeyMultibase`.
```
Comment (dlongley) [inline CHANGELOG.md:6]:
```suggestion
- `AsymmetricKey` now sets `algorithm` using the prefix of the `publicKeyMultibase`.
```

File: lib/AsymmetricKey.js [hunk 3]
```diff
@@ -105,6 +106,35 @@ export class AsymmetricKey {
...
+    // @see https://github.com/multiformats/multicodec/blob/master/table.csv
+    // maps the bs58 prefix to the JOSE algorithm name
+    const algorithms = new Map([
+      ['6Mk', 'Ed25519'],
+      ['Dna', 'P-256'],
+      ['82L', 'P-384'],
+      ['2J9', 'P-521'],
+      // FIXME this prefix was not generated by an ed448 did key implementation
+      ['6GP', 'Ed448'],
+      ['Q3s', 'secp256k1']
```
Comment (dlongley) [inline lib/AsymmetricKey.js:129]: These should be prefixed with `z` to ensure there's no cross-encoding mix up.

```suggestion
      ['z6Mk', 'Ed25519'],
      ['zDna', 'P-256'],
      ['z82L', 'P-384'],
      ['z2J9', 'P-521'],
      // FIXME this prefix was not generated by an ed448 did key implementation
      ['z6GP', 'Ed448'],
      ['zQ3s', 'secp256k1']
```

File: lib/AsymmetricKey.js [hunk 4]
```diff
@@ -105,6 +106,35 @@ export class AsymmetricKey {
...
+    ]);
+    const multibaseBs58Header = 'z';
+    if(!publicKeyMultibase.startsWith(multibaseBs58Header)) {
+      return;
+    }
```
Comment (dlongley) [inline lib/AsymmetricKey.js:134]:
```suggestion
```

File: lib/AsymmetricKey.js [hunk 5]
```diff
@@ -105,6 +106,35 @@ export class AsymmetricKey {
...
+    return algorithms.get(publicKeyMultibase.substr(1, 3));
```
Comment (dlongley) [inline lib/AsymmetricKey.js:135]:
```suggestion
    return algorithms.get(publicKeyMultibase?.slice(0, 3));
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/AsymmetricKey.js (hunk 1 above)
Comment (aljones15) [reply lib/AsymmetricKey.js]: Addressed here: https://github.com/digitalbazaar/webkms-client/pull/42/commits/b2b6f51cb5738d003a1b2b89ea4a2f6192cd397d , but do we want a `constats.js` module to put this in as we might want this functionality in other key classes?

**Pull request merged**